client/src/core/props-behavior.js

```diff
@@ -131,107 +131,148 @@ class PropsSelector {
   }
 
   /**
-   * Select a prop for building mode
-   * @param {Vector2} pointer - 2D Pointer for the raycaster.
-   * @param {boolean} add - True to add the newfound prop to the existing
-   *                        selection list, false (default) to clear the
-   *                        list first.
+   * Traverse up the object hierarchy to find the root prop object.
+   *
+   * @param {Object3D} object - The child object to start from.
+   * @returns {Object3D} The root object containing userData.prop,
+   *                           or the original object if none found.
    */
-  select(pointer, add = false) {
-    let done = false;
+  getRootPropObject(object) {
+    while (object && !object.userData?.prop && object.parent) {
+      object = object.parent;
+    }
+    return object;
+  }
 
-    this.clickRaycaster.setFromCamera(
-        pointer, this.engine3d.camera,
-    );
+  /**
+  * Add a single root prop to the current selection list.
+  *
+  * @param {Object3D} rootProp - Root object representing the prop.
+  */
+  selectSingleProp(rootProp) {
+    const boundingBox = rootProp.getObjectByName(boundingBoxName)?.clone();
+    if (!boundingBox) return;
+
+    const {x, y, z} = rootProp.userData.prop;
+
+    const stagingProp = rootProp.clone();
+    stagingProp.visible = true;
+    stagingProp.position.set(x, y, z);
+
+    this.engine3d.appendToNode(this.stagingNode, stagingProp);
+    stagingProp.updateMatrix();
+
+    boundingBox.position.set(x, y, z);
+    boundingBox.rotation.copy(rootProp.rotation);
+    boundingBox.updateMatrix();
+    boundingBox.visible = true;
+    this.engine3d.addHelperObject(boundingBox);
+
+    this.props.push({prop: rootProp, stagingProp, boundingBox});
+  }
 
-    const intersects =
-        this.clickRaycaster.intersectObjects(
-            this.engine3d.getSelectableNodes(), true,
+  /**
+  * Select all props in the same chunk within a given radius.
+  *
+  * @param {Object3D} rootProp - Root prop used as center point.
+  * @param {number} radius - Distance (in meters) to include nearby props.
+  */
+  selectNearbyProps(baseProp) {
+    const chunkHandle = baseProp.userData.chunkNodeHandle;
+    if (!chunkHandle) return;
+
+    const chunkId = this.worldManager.chunkKeys.get(chunkHandle);
+    if (!chunkId) return;
+
+    // Find all props in the same chunk
+    // eslint-disable-next-line
+    for (const [propId, obj3d] of this.worldManager.props.entries()) {
+      if (
+        obj3d.userData.chunkNodeHandle === chunkHandle &&
+        obj3d !== baseProp &&
+        obj3d.visible
```

Comment:
Does it even need to be visible? I don't know/remember how AW actually behaves in that regard